shm/ship.cpp

```diff
@@ -1,20 +1,64 @@
 #include "ship.hpp"
-
+#include <iostream>
 #include <algorithm>
 #include <numeric>
 
 Ship::Ship()
     : id_(-1) {}
 
-Ship::Ship(size_t capacity, size_t maxCrew, size_t speed, const std::string& name, int id)
-    : capacity_(capacity), maxCrew_(maxCrew), crew_(0), speed_(speed), name_(name), id_(id) {}
+Ship::Ship(size_t capacity, size_t maxCrew, size_t speed, const std::string& name, int id, std::shared_ptr<Time> publisher)
+    : capacity_(capacity), 
+    maxCrew_(maxCrew), 
+    crew_(0), 
+    speed_(speed), 
+    name_(name), 
+    id_(id),
+    publisher_(publisher){
+    this->publisher_->addObserver(this);
+    }
 
-Ship::Ship(size_t maxCrew, size_t speed, int id)
-    : Ship(0, maxCrew, speed, "", id) {}
+//Ship::Ship(size_t maxCrew, size_t speed, int id)
+//   : Ship(0, maxCrew, speed, "", id) {}
 
 void Ship::setName(const std::string& name) {
     name_ = name;
 }
+   //wziąć pod uwagę available space - countFreeSpace w klasie player!
+    //porównanie dwóch towarów - akumulacja tych samych typów towarów
+    //czy zmiesci sie caly towar (maxcapacity?)
+
+// void Ship::unload(Cargo* cargo) {
+//     auto matchCargo = find_if(cargo_.begin(), cargo_.end(), [cargo](const auto& ptr) {
+//                  return ptr.get() == cargo;
+//                  });
+//     if (matchCargo != cargo_.end())
+//         cargo_.erase(matchCargo);
+// }
+  //sprawdzić ilość czy jest 0 - usuwamy
+  //ile zostanie towaru
```

Comment:
Why have you readded this comment? We do not need unload since if we remove item by either -= item or using removeCargo.